### PR TITLE
PT-2258: Update Catalog module

### PR DESCRIPTION
**Pull request by aizhuli**
## Description
Update Catalog module to implement Generic CRUD Services

### CODE REVIEW

File: src/VirtoCommerce.CatalogModule.BulkActions/VirtoCommerce.CatalogModule.BulkActions.csproj [hunk 1]
```diff
@@ -22,8 +22,8 @@
 
     <ItemGroup>
         <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All" />
-        <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="3.1.8" />
-        <PackageReference Include="VirtoCommerce.BulkActionsModule.Core" Version="3.1.0" />
+        <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="3.1.18" />
```
Comment (eMazeika) [inline src/VirtoCommerce.CatalogModule.BulkActions/VirtoCommerce.CatalogModule.BulkActions.csproj:25]: current Platform references 
<PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="3.1.13" />
and module can't reference higher version then Platform does. Sync them.

File: tests/VirtoCommerce.CatalogModule.Tests/PropertyDictionaryItemTest.cs [hunk 2]
```diff
@@ -16,8 +17,9 @@ public class PropertyDictionaryItemTest
         [Fact]
         public async Task Add_New_DictionaryItems_To_Property_And_Then_Choose_DictItem_For_Product()
         {
-            var propDictionaryService = new Mock<IPropertyDictionaryItemService>().Object;
-            var propDictionarySearchService = new Mock<IPropertyDictionaryItemSearchService>();
+            var propDictionaryService = new Mock<IPropertyDictionaryItemService>().As<ICrudService<PropertyDictionaryItem>>().Object;
+            var propDictionarySearchService = new Mock<IPropertyDictionaryItemSearchService>().As<ISearchService<PropertyDictionaryItemSearchCriteria, PropertyDictionaryItemSearchResult, PropertyDictionaryItem>>();
+            var productCrudService = new Mock<IItemService>().As<ICrudService<CatalogProduct>>();
```
Comment (eMazeika) [inline tests/VirtoCommerce.CatalogModule.Tests/PropertyDictionaryItemTest.cs:22]: rename to follow single naming style

File: src/VirtoCommerce.CatalogModule.Web/VirtoCommerce.CatalogModule.Web.csproj [hunk 3]
```diff
@@ -23,8 +23,8 @@
         <NoWarn>1701;1702;1705;1591</NoWarn>
     </PropertyGroup>
     <ItemGroup>
-        <PackageReference Include="VirtoCommerce.Platform.Core" Version="3.57.0" />
-        <PackageReference Include="VirtoCommerce.Platform.Data" Version="3.57.0" />
+        <PackageReference Include="VirtoCommerce.Platform.Core" Version="3.62.0" />
+        <PackageReference Include="VirtoCommerce.Platform.Data" Version="3.62.0" />
         <PackageReference Include="VirtoCommerce.Platform.Security" Version="3.57.0" />
```
Comment (eMazeika) [inline src/VirtoCommerce.CatalogModule.Web/VirtoCommerce.CatalogModule.Web.csproj:28]: the same version of Platform should be referenced (3.62)

File: src/VirtoCommerce.CatalogModule.Data/VirtoCommerce.CatalogModule.Data.csproj [hunk 4]
```diff
@@ -33,8 +33,8 @@
         <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="3.1.8" />
         <PackageReference Include="VirtoCommerce.CoreModule.Core" Version="3.1.0" />
         <PackageReference Include="VirtoCommerce.ExportModule.Data" Version="3.4.0" />
-        <PackageReference Include="VirtoCommerce.Platform.Core" Version="3.57.0" />
-        <PackageReference Include="VirtoCommerce.Platform.Data" Version="3.57.0" />
+        <PackageReference Include="VirtoCommerce.Platform.Core" Version="3.62.0" />
+        <PackageReference Include="VirtoCommerce.Platform.Data" Version="3.62.0" />
         <PackageReference Include="VirtoCommerce.Platform.Security" Version="3.57.0" />
```
Comment (eMazeika) [inline src/VirtoCommerce.CatalogModule.Data/VirtoCommerce.CatalogModule.Data.csproj:38]: sync versions

File: src/VirtoCommerce.CatalogModule.Data/Services/CategoryService.cs [hunk 5]
```diff
@@ -1,261 +1,234 @@
...
+using VirtoCommerce.Platform.Data.GenericCrud;
+using VirtoCommerce.Platform.Core.GenericCrud;
+
+namespace VirtoCommerce.CatalogModule.Data.Services
+{
+    public class CategoryService : CrudService<Category, CategoryEntity, CategoryChangingEvent, CategoryChangedEvent>, ICategoryService
+    {
+        private new readonly IPlatformMemoryCache _platformMemoryCache;
+        private new readonly Func<ICatalogRepository> _repositoryFactory;
+        private new readonly IEventPublisher _eventPublisher;
```
Comment (eMazeika) [inline src/VirtoCommerce.CatalogModule.Data/Services/CategoryService.cs:29]: consider removing all "private new" and using inherited members

File: src/VirtoCommerce.CatalogModule.Data/Services/ItemService.cs [hunk 6]
```diff
@@ -15,18 +15,20 @@
...
-        private readonly IEventPublisher _eventPublisher;
+        private new readonly Func<ICatalogRepository> _repositoryFactory;
+        private new readonly IEventPublisher _eventPublisher;
         private readonly AbstractValidator<IHasProperties> _hasPropertyValidator;
-        private readonly ICatalogService _catalogService;
+        private readonly ICrudService<Catalog> _catalogService;
         private readonly ICategoryService _categoryService;
         private readonly IOutlineService _outlineService;
-        private readonly IPlatformMemoryCache _platformMemoryCache;
+        private new readonly IPlatformMemoryCache _platformMemoryCache;
```
Comment (eMazeika) [inline src/VirtoCommerce.CatalogModule.Data/Services/ItemService.cs:31]: consider removing all "private new" and using inherited members

File: src/VirtoCommerce.CatalogModule.Data/Services/PropertyDictionaryItemService.cs [hunk 7]
```diff
@@ -1,88 +1,76 @@
...
 
 namespace VirtoCommerce.CatalogModule.Data.Services
 {
-    public class PropertyDictionaryItemService : IPropertyDictionaryItemService
+    public class PropertyDictionaryItemService : CrudService<PropertyDictionaryItem, PropertyDictionaryItemEntity, PropertyDictionaryItemChangingEvent, PropertyDictionaryItemChangedEvent>, IPropertyDictionaryItemService
     {
-        private readonly Func<ICatalogRepository> _repositoryFactory;
-        private readonly IPlatformMemoryCache _platformMemoryCache;
+        private new readonly Func<ICatalogRepository> _repositoryFactory;
+        private new readonly IPlatformMemoryCache _platformMemoryCache;
```
Comment (eMazeika) [inline src/VirtoCommerce.CatalogModule.Data/Services/PropertyDictionaryItemService.cs:23]: consider removing all "private new" and using inherited members

**Review by eMazeika — CHANGES_REQUESTED** [review]
check sonar comments as well

File: src/VirtoCommerce.CatalogModule.Data/Repositories/CatalogRepository.cs [hunk 8]
```diff
@@ -1,721 +1,724 @@
...
+
+                var categoryPropertiesIds = await Properties.Where(x => categoriesIds.Contains(x.CategoryId))
+                                                            .Select(x => x.Id).ToArrayAsync();
+                await GetPropertiesByIdsAsync(categoryPropertiesIds);
+            }
+
+            return result;
+        }
+
+        public virtual async Task<IEnumerable<ItemEntity>> GetItemByIdsAsync(IEnumerable<string> itemIds, string responseGroup = null)
```
Comment (ksavosteev) [inline src/VirtoCommerce.CatalogModule.Data/Repositories/CatalogRepository.cs:120]: DISALLOWED: Changing the type of a member

**Comment by akak1977** [comment]
Closed for reason of priority PR:
https://github.com/VirtoCommerce/vc-module-catalog/pull/612

**Pull request closed without merging**